bindings/python/pytpm2tss/__init__.py

```diff
@@ -0,0 +1,2533 @@
+"""
+SPDX-License-Identifier: BSD-2
+
+Copyright 2018, Fraunhofer SIT
+All rights reserved.
+"""
+
+from . import headers
+from cffi import FFI
+
+_ffi = FFI()
+
+_ffi.cdef(headers.tss2_common_h)
+_ffi.cdef(headers.tss2_tpm2_types_h)
+_ffi.cdef(headers.tss2_tcti_h)
+_ffi.cdef(headers.tss2_esys_h)
```

Comment:
This would probably mean splitting this pytpm2tss out into its own project, right ?

I was hoping to see pytpm2tss (and other bindings) and the tpm2tss libraries live in the same project, so we can advance the APIs together.

Any idea how we could achieve this ?
And if I put the _cffi-constructors into a Makefile-rule; what would be the resulting files that I need to install to where or would it be the .pyc files ? (not enough knowledge here)

Reply:
I reworked this a lot with the build.py and the setup.py.
Still needs a lot of work though

Reply:
> I was hoping to see pytpm2tss (and other bindings) and the tpm2tss libraries live in the same project, so we can advance the APIs together.

Sounds nice from a maintainer's point of view, but I'm not sure whether that will work for the users. The distributions will ship some version of the C library and keep it stable for some months/years. Eventually, they might also ship the Python bindings, though I'd expect that to be less common (since you can usually install such things easily from PyPI). But even if they do, at some point the users probably want to upgrade to the newest Python bindings to get the latest features/fixes. So ideally the Python bindings will always support all versions of the C libraries that were in use during the last couple of years, and hide all backwards-incompatible changes as far as possible.

> And if I put the _cffi-constructors into a Makefile-rule; what would be the resulting files that I need to install to where or would it be the .pyc files ? (not enough knowledge here)

It seems you already found some of the setup.py magic :) I'll look at those changes later.

Reply:
I found this: https://blog.kevin-brown.com/programming/2014/09/24/combining-autotools-and-setuptools.html
We'll see, how that magic works.
I'm still struggling to get the builddir concept mapped to setup.py in my head...